### PR TITLE
test/e2e/auth: fix audit log test format parsing

### PR DESCRIPTION
**Pull request by ericchiang**
Fixes https://github.com/kubernetes/kubernetes/issues/51556

```release-note
NONE
```

cc @CaoShuFeng

Still need to figure out how to run this test locally.

### CODE REVIEW

File: test/e2e/auth/audit.go [hunk 1]
```diff
@@ -138,6 +139,26 @@ func expectAuditLines(f *framework.Framework, expected []auditEvent) {
 }
 
 func parseAuditLine(line string) (auditEvent, error) {
+	var e struct {
```
Comment (liggitt) [inline test/e2e/auth/audit.go]: Is this struct not already defined in API types we can reference?

Comment (ericchiang) [reply test/e2e/auth/audit.go]: Updated to use the audit Event type.

File: test/e2e/auth/audit.go [hunk 2]
```diff
@@ -126,9 +124,6 @@ func expectAuditLines(f *framework.Framework, expected []auditEvent) {
 		if _, found := expectations[event]; found {
 			expectations[event] = true
 		}
-
-		// /version should not be audited (filtered in the policy).
-		Expect(event.uri).NotTo(HavePrefix("/version"))
```
Comment (ericchiang) [inline test/e2e/auth/audit.go]: Removed since non-resource URL requests are now tracked.

**Comment by ericchiang** [comment]
Managed to run this locally with some local-up modifications and modifying the test to read a file on disk

```
[eric@localhost e2e]$ ginkgo -- --provider=local '--ginkgo.focus=\[Feature:Audit\]' --ginkgo.v --kubeconfig=/var/run/kubernetes/admin.kubeconfig
2017/08/29 17:02:39 proto: duplicate proto type registered: google.protobuf.Any
2017/08/29 17:02:39 proto: duplicate proto type registered: google.protobuf.Duration
2017/08/29 17:02:39 proto: duplicate proto type registered: google.protobuf.Timestamp
Aug 29 17:02:39.677: INFO: Overriding default scale value of zero to 1
Aug 29 17:02:39.677: INFO: Overriding default milliseconds value of zero to 5000
=== RUN   TestE2E
I0829 17:02:39.790684   27185 e2e.go:363] Starting e2e run "899fdff8-8d16-11e7-9443-54ee7553240e" on Ginkgo node 1
Running Suite: Kubernetes e2e suite
===================================
Random Seed: 1504051359 - Will randomize all specs
Will run 1 of 682 specs

Aug 29 17:02:39.833: INFO: >>> kubeConfig: /var/run/kubernetes/admin.kubeconfig
Aug 29 17:02:39.842: INFO: Waiting up to 4h0m0s for all (but 0) nodes to be schedulable
Aug 29 17:02:39.859: INFO: Waiting up to 10m0s for all pods (need at least 0) in namespace 'kube-system' to be running and ready
Aug 29 17:02:39.867: INFO: 1 / 1 pods in namespace 'kube-system' are running and ready (0 seconds elapsed)
Aug 29 17:02:39.867: INFO: expected 1 pod replicas in namespace 'kube-system', 1 are Running and Ready.
Aug 29 17:02:39.872: INFO: Waiting for pods to enter Success, but no pods in "kube-system" match label map[name:e2e-image-puller]
Aug 29 17:02:39.872: INFO: Dumping network health container logs from all nodes...
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[sig-auth] Advanced Audit [Feature:Audit] 
  should audit API calls
  /home/eric/src/k8s.io/kubernetes/test/e2e/auth/audit.go:38
[BeforeEach] [sig-auth] Advanced Audit [Feature:Audit]
  /home/eric/src/k8s.io/kubernetes/test/e2e/framework/framework.go:141
STEP: Creating a kubernetes client
Aug 29 17:02:39.880: INFO: >>> kubeConfig: /var/run/kubernetes/admin.kubeconfig
STEP: Building a namespace api object
STEP: Waiting for a default service account to be provisioned in namespace
[It] should audit API calls
  /home/eric/src/k8s.io/kubernetes/test/e2e/auth/audit.go:38
Aug 29 17:02:43.930: INFO: Waiting for pod audit-pod to disappear
Aug 29 17:02:43.933: INFO: Pod audit-pod still exists
Aug 29 17:02:45.933: INFO: Waiting for pod audit-pod to disappear
Aug 29 17:02:45.936: INFO: Pod audit-pod still exists
Aug 29 17:02:47.933: INFO: Waiting for pod audit-pod to disappear
Aug 29 17:02:47.935: INFO: Pod audit-pod no longer exists
[AfterEach] [sig-auth] Advanced Audit [Feature:Audit]
  /home/eric/src/k8s.io/kubernetes/test/e2e/framework/framework.go:142
Aug 29 17:02:48.388: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "e2e-tests-audit-f9745" for this suite.
Aug 29 17:02:54.437: INFO: namespace: e2e-tests-audit-f9745, resource: bindings, ignored listing per whitelist
Aug 29 17:02:54.498: INFO: namespace e2e-tests-audit-f9745 deletion completed in 6.108438895s

• [SLOW TEST:14.618 seconds]
[sig-auth] Advanced Audit [Feature:Audit]
/home/eric/src/k8s.io/kubernetes/test/e2e/auth/framework.go:22
  should audit API calls
  /home/eric/src/k8s.io/kubernetes/test/e2e/auth/audit.go:38
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSAug 29 17:02:54.499: INFO: Running AfterSuite actions on all node
Aug 29 17:02:54.499: INFO: Running AfterSuite actions on node 1

Ran 1 of 682 Specs in 14.666 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 681 Skipped --- PASS: TestE2E (14.71s)
PASS
```

**Comment by liggitt** [comment]
LGTM. @sttts?

**Comment by CaoShuFeng** [comment]
If we are in a hurry to fix this we can add `--audit-log-format=legacy` to api-server.
But I don't know where to put it. ref: https://github.com/kubernetes/kubernetes/pull/50971#issuecomment-323669509

**Comment by ericchiang** [comment]
I think this fix is fine. It supports both formats now, so it'll work regardless of the setup, which seems more ideal.

**Comment by CaoShuFeng** [comment]
> I think this fix is fine. It supports both formats now, so it'll work regardless of the setup, which seems more ideal.

Agree! Thanks.

**Pull request merged**